astroid/brain/brain_type.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+"""
+Astroid hooks for type support.
+
+Starting from python3.9, type object behaves as it had __class_getitem__ method.
+However it was not possible to simply add this method inside type's body, otherwise
+all types would also have this method. In this case it would have been possible
+to write str[int].
+Guido Van Rossum proposed a hack to handle this in the interpreter:
+https://github.com/python/cpython/blob/master/Objects/abstract.c#L186-L189
+
+This brain follows the same logic. It is no wise to add permanently the __class_getitem__ method 
+to the type object. Instead we choose to add it only in the case of a subscript node
+which inside name node is type. 
+Doing this type[int] is allowed whereas str[int] is not.
+
+Thanks to Lukasz Langa for fruitful discussion.
+"""
+import sys
+
+from astroid import MANAGER, extract_node, inference_tip, nodes
+
+
+PY39 = sys.version_info >= (3, 9)
+
+
+def _looks_like_type_subscript(node):
```

Comment:
I think we can add typing directly in signature as we do not have python 3.5 to take into consideration anymore.
```suggestion
def _looks_like_type_subscript(node: nodes.Name) -> bool:
```

Reply:
@Pierre-Sassoulas, sadly we did not yet throw away `python3.5`...
#864 has not been merged yet. I wanted to have @PCManticore opinion about it (i.e should the drop of `python3.5` be part of the next release or should it be delayed to the one after..)
I hope it will be done soon.

Reply:
Ho ok, sorry :)

Reply:
Hey sorry, don't hold PRs off because of me. I might not be able to respond in a timely manner. Feel free to drop Python 3.5 if the Python version is already EOLed.